quamash/__init__.py

```diff
@@ -112,17 +81,23 @@ class QThreadExecutor(QtCore.QObject):
 	Same API as `concurrent.futures.Executor`
 
 	>>> from quamash import QThreadExecutor
-	>>> with QThreadExecutor(5) as executor:
+	>>> QtCore = getfixture('qtcore')
+	>>> with QThreadExecutor(QtCore.QThread, 5) as executor:
 	...     f = executor.submit(lambda x: 2 + x, 2)
 	...     r = f.result()
 	...     assert r == 4
 	"""
 
-	def __init__(self, max_workers=10, parent=None):
-		super().__init__(parent)
+	def __init__(self, qthread_class, max_workers=10):
+		super().__init__()
+		assert isinstance(qthread_class, type)
+
+		class QThreadWorker(_QThreadWorker, qthread_class):
```

Comment:
This is a bit ugly, maybe we could have a small nested wrapper class that calls _QThreadWorker instead? I.e., composition rather than inheritance.